openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1119,6 +1121,67 @@ Cache-Control: no-store
 }
 ```
 
+# Callback Endpoint {#callback_endpoint}
+
+This endpoint is used to receive notification from the Wallet whether credential has been successfully received in the Wallet or not. The Credential Issuer needs to return `callback_id` in the Credential Response or a Batch Credential Response for the Wallet to be able to use this Endpoint. Support for this endpoint is OPTIONAL.
+
+This endpoint can be used after the Credential Issuer has sent Credential Response or Batch Credential Response. It enables the Credential Issuer to take subsequent actions after issuance, depending on whether the credential has been accepted and successully stored by the Wallet, rejected by the Wallet, or errors and other unforeseen circumstances have occurred during the Wallet's processing.
+
+Communication with the Callback Endpoint MUST utilize TLS.
+
+## Callback from the Wallet {#callback}
+
+The Wallet sends an HTTP POST request to the Callback Endpoint with the following parameters in the entity-body and using the `application/json` media type.
+
+* `issuer_state`: REQUIRED if the `issuer_state` parameter was present in the Credential Offer (see (#credential_offer)).
+* `credentials`: A JSON array of objects, where each object consists of the following parameters:
+  * `callback_id`: REQUIED. A JSON string received in Credential Response or Batch Credential Response.
+  * `status`: REQUIRED. Status whether the credential issuance was successful or not. The value MUST be either `success` or `error`.
```

Comment:
I think `error` is intended to be used for all non-success cases, including cases where no error occurred but the user or wallet decided not to accept a valid credential?

```suggestion
  * `status`: REQUIRED. Status whether the credential issuance was successful or not. The value MUST be either `success` or `failure`.
```

I'd also be okay with keeping 'error' (there's precedent in RFC6749 for using error in these cases) but just adding a bit of text saying "`error` indicates any case where the wallet has not successfully stored the credential, for example a failure to communicate with the wallet's backend, a problem with the credential, or the wallet or user may have decided not to accept a valid credential. The failure may be temporary or permanent."

Reply:
Would `accepted` and `not_accepted` (or `rejected`) better describe the semantics of this status, namely because a credential may be not accepted due to an user decision and not due to an error?

Reply:
We had `accepted`, `rejected` and `error` but refined that down to the current two on the argument that many wallets will not offer a choice to accept/reject a credential at the point of credential response. The argument being that the wallet has already given explicit consent it wants the credential(s) by completing the rest of the flow, so it will simply absorb it if it's valid without explicit interaction on the device at this stage in the flow.

I would prefer we accomodate for wallets that _do_ want that explicit UX at this final stage, even if they're not currently commonly implemented, so I would prefer `accepted` and `rejected` for exactly this use-case.

Reply:
I am ok with `accepted` and `rejected`. I think Giuseppe was asking if this better be a boolean, but sounds like strings work just fine..?

Reply:
```suggestion
  * `status`: REQUIRED. Status whether the credential issuance was successful or not. The value MUST be either `accepted` or `rejected`.
```